src/python/pants/engine/legacy/structs.py

```diff
@@ -288,6 +293,8 @@ def legacy_globs_class(self):
     """The corresponding `wrapped_globs` class for this BaseGlobs."""
 
   def __init__(self, *patterns, **kwargs):
+    self._patterns = patterns
+    self._kwargs = kwargs
```

Comment:
Since this is mutated below, it will not end up containing anything interesting. Can fix in a followup, but IMO, attempting to reproduce the syntax here is probably overkill.